packages/time-picker/src/TimePicker/utils/convertTimeToZonedDateTime.ts

```diff
@@ -0,0 +1,26 @@
+import { now, ZonedDateTime } from "@internationalized/date"
+
+export const convertTimeToZonedDateTime = ({
+  date,
+  hour,
+  minutes,
+  timeZone,
+}: {
+  date?: Date
+  hour: number
+  minutes: number
+  timeZone: string
+}): ZonedDateTime => {
+  // using this instead of new Date().getTimezoneOffset as ZonedDateTime constructor measures offset different
+  const today = now(timeZone)
+  const todayDate = today.toDate()
+  return new ZonedDateTime(
+    (date ?? todayDate).getFullYear(),
+    (date ?? todayDate).getMonth() + 1,
+    (date ?? todayDate).getDate(),
+    timeZone,
+    today.offset,
```

Comment:
Won’t the offset be dependent on the passed date, not on the offset of today?

Reply:
We do generate the offset from the passed in timezone, but maybe I can assign a variable name to this so it's less confusing.